src/net.cpp

```diff
@@ -2642,17 +2651,140 @@ void CConnman::RecordBytesRecv(uint64_t bytes)
     nTotalBytesRecv += bytes;
 }
 
-void CConnman::RecordBytesRecvByMsgType(mapMsgTypeSize msgtype_bytes)
+void CConnman::RecordMsgStatsRecv(std::map<std::string, std::tuple<int, uint64_t>> msgtype_countbytes,
+                                  ConnectionType conn_type, Network net_type)
 {
-    for (auto const& msg_type_bytes : msgtype_bytes) {
-        auto i = m_msgtype_bytes_recv.find(msg_type_bytes.first);
+    for (auto const& msgtype_stats : msgtype_countbytes) {
+        std::string msg_type = msgtype_stats.first;
+        auto i = m_msgtype_bytes_recv.find(msg_type);
         if (i == m_msgtype_bytes_recv.end()) {
             i = m_msgtype_bytes_recv.find(NET_MESSAGE_TYPE_OTHER);
         }
 
         assert(i != m_msgtype_bytes_recv.end());
-        i->second += msg_type_bytes.second;
+        std::tuple<int, uint64_t> count_bytes = msgtype_stats.second;
+
+        int msg_count = std::get<0>(count_bytes);
+        uint64_t num_bytes = std::get<1>(count_bytes);
+
+        if (num_bytes > 0) {
+            i->second += num_bytes;
+
+            NetMsgStatsKey stats_key = {msg_type, conn_type, net_type};
+
+            // LogPrint(BCLog::NET, "\nstacie - incremeting received message count by %d, num_bytes by %d for (%s, %s, %s)\n",
+            //          msg_count, num_bytes, msg_type, ConnectionTypeAsString(conn_type), NetworkAsString(net_type));
+            auto stats_iterator = m_netmsg_stats_recv.find(stats_key);
+
+            // If this stat does not exist, add it (could also initialize all stats to zero?)
+            if (stats_iterator == m_netmsg_stats_recv.end()) {
+                NetMsgStatsValue stats_value = {msg_count, num_bytes};
+                m_netmsg_stats_recv.insert(std::map<NetMsgStatsKey, NetMsgStatsValue>::value_type(stats_key, stats_value));
+            } else {
+                // otherwise, update the message count and number of bytes
+                stats_iterator->second.msg_count += msg_count;
+                stats_iterator->second.num_bytes += num_bytes;
+            }
+        }
+    }
+}
+
+std::map<std::string, CConnman::NetMsgStatsValue> CConnman::AggregateNetMsgStats(const std::map<NetMsgStatsKey, NetMsgStatsValue>& raw_stats, std::vector<int> filters) const
+{
+    // An object for the results
+    // The depth of the return object depends on the number of filters
+    // ideally this would be JSON but that appears to be hard to do in C++
+    std::map<std::string, NetMsgStatsValue> aggregate_stats = {};
+
+    int num_filters = 0;
+    if (sizeof(filters) > 0) {
+        sizeof(filters) / sizeof(filters[0]);
+    }
```

Comment:
Maybe you are looking for `const size_t num_filters = filters.size();`?

Reply:
yes! :sob: I just discovered this bug independently and fixed. Should have checked the comments here first :sweat_smile: 

Resolved in https://github.com/satsie/bitcoin/pull/3/commits/765bdd48f157ecfb7a36de530db79ce9cb0fafe0#diff-00021eed586a482abdb09d6cdada1d90115abe988a91421851960e26658bed02R2699